### PR TITLE
Migrate to v2 addon format

### PR DESCRIPTION
**Pull request by dfreeman**
This is the first of what will likely be a sequence of breaking-change PRs leading up to `ember-exclaim@2.0`. The goal is to be as un-disruptive as possible in what we actually break, while still taking opportunities to modernize the codebase and improve overall DX/performance.

Notable breaking changes in this PR:
 - dropping Ember < 3.28 from our test matrix
 - requiring Embroider or `ember-auto-import`@2 in the consuming package, as we're now a v2 addon

Note that the general structure for v2 addons is a monorepo with one workspace for the addon itself and another for a test application (what used to be the "dummy app"). The addon is "just" an npm package that uses Rollup for its build process, while the test app is a standard Ember application with none of the weirdness associated with the hybrid `EmberAddon`/dummy application setup from v1 addons.

This PR also introduces a third `playground-app` workspace, which contains the stuff we deploy to GitHub pages to demo Exclaim. One nice side effect of the v2 setup is we don't have to force our tests and docs/demos/etc to coexist in a single application.

I've done my best to break things out by commit, but unfortunately the first one in particular is large and a bit entangled.
 - 3fd75e0df23e36d82967c8f37aef85d02deeec42: migrate into the v2 addon structure generated by https://github.com/embroider-build/addon-blueprint
   - most of the updates here either boilerplate generated code or file moves, but a few things were manual changes I'll call out
   - internal import paths were updated to be relative to adhere to standard Node/bundler resolution rules
   - migrating from `assert.equal` to `assert.strictEqual` in tests—this shouldn't have had to be part of this commit, but ended up being difficult to extract after the fact
 - 49c06d49c13553b044e8f69e16aae1e05e35b39a: turn off some lint rules that guard against the use of old Ember APIs. We definitely want to fix these, but not as part of this already-large PR.
 - 9f3292b25902ecd732e414c251e0913447a3bfdf: fix imports from `@ember/string` that moved to `@ember/template` a long long time ago
 - 94a58cd16594e188f893174875e4742a6806652a: clean up a few unneeded dependencies from the `test-app` workspace
 - c24dc87bb2d73e6573ccc07d367a3ae6fe173898: split the demo routes out of `test-app` into a separate `playground-app` workspace so that we don't need to maintain dependencies (and juggle ember-try scenarios) for both things together
 - b9927a21cc6b8f863da6bc6b0d3599d705956b1c: upgrade to `ember-ace` v3, which is much easier to use in modern apps since it cleans up use of some old Ember APIs and is itself a v2 addon that makes use of standard bundler features now instead of custom `ember-cli-build.js` configuration
 - 508d2ed97287ece91dff0c55ad2f650a0c6f30a5: clean up our ember-try scenarios and get CI passing
 - abe5857ef2daa7e1f2f4a3a69f14c766850c77b4: add our standard "Salsify FE team" automated release configuration

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,56 +1,77 @@
...
+          cache: yarn
       - name: Install Dependencies
         run: yarn install --frozen-lockfile
       - name: Lint
-        run: yarn lint
-      - name: Test
-        run: yarn test:ember
+        run: yarn run lint
+      - name: Run Tests
+        run: yarn run test
```
Comment (katrynmc) [inline .github/workflows/ci.yml:30]: [nit] There's some inconsistency in your invocation of the test command here and below

File: LICENSE.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2019
+Copyright (c) 2023
```
Comment (katrynmc) [inline LICENSE.md:3]: lol time flies

File: ember-exclaim/package.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  "license": "MIT",
+  "author": "Dan Freeman",
+  "files": [
+    "addon-main.cjs",
+    "declarations",
+    "dist"
+  ],
+  "scripts": {
+    "build": "rollup --config",
+    "lint": "concurrently 'npm:lint:*(!fix)' --names 'lint:'",
```
Comment (katrynmc) [inline ember-exclaim/package.json:22]: Oh neat I was not familiar with concurrently

**Review by katrynmc — APPROVED** [review]

File: ember-exclaim/package.json (hunk 3 above)
Comment (dfreeman) [reply ember-exclaim/package.json:22]: Tbh I find the intermingled output a little rocky sometimes, but it's better than having to spin up separate shells and (more importantly) it's what the blueprint does by default, so I went with it

**Pull request merged**